app/controllers/admin/vendors_controller.rb

```diff
@@ -55,26 +34,22 @@ def index
   def edit
     # Get vendor so form knows to make submit button say "Update Vendor"
     @vendor = Vendor.find(params[:id])
+    # Pass in params from form if redirected from #update
+    if flash[:prev_params]
+      @vendor.assign_attributes(flash[:prev_params])
+    end
   end
 
   def update
     vendor = Vendor.find(params[:id])
     success = vendor.update_attributes(vendor_params)
-
-    if success
-      vendor_success("update")
-    else
-      vendor_fail("update", vendor)
-    end
+    verify_and_redirect(success, vendor, admin_vendors_path, vendor_params)
   end
 
   def destroy
     vendor = Vendor.find(params[:id])
     success = vendor.destroy
-    if success
-      vendor_success("delete")
-    else
-      vendor_fail("delete", vendor)
-    end
+    puts 'got here!'
```

Comment:
I think this line should be removed. I suppose it was for testing.

Reply:
Should be fixed now.